src/ui/pages/equipment/asset/Asset.js

```diff
@@ -335,9 +338,38 @@ export default class Asset extends Entity {
                 column: 2,
                 order: 11
             },
+            {
+                id : 'PARTCUSTOMFIELDS',
+                label : 'Part Custom Fields',
+                isVisibleWhenNewEntity: false,
+                maximizable: false,
+                customVisibility: () => this.state.part,
+                render: () => 
+                    <CustomFields 
+                        children={this.children}
+                        entityCode='PART'
+                        entityKeyCode={this.state.part && this.state.part.code}
+                        classCode={this.state.part && this.state.part.customField}
+                        customFields={layout.partCustomField && layout.partCustomField}
+                        updateEntityProperty={this.updateEntityProperty.bind(this)}
+                        readonly={true}/>
+                ,
+                column: 2,
+                order: 12
+            }
         ]
     }
 
+    setAssetPart = partCode=> {
+        return WSParts.getPart(partCode).then(response => {
+            this.setState({part : response.body.data})
+            this.setLayout({partCustomField : response.body.data.customField})
```

Comment:
When accessing the custom fields for the part, these 2 lines make it so that both `this.state.part.customField` or `this.state.layout.customField` can be used, with no clear preference, which is a source of ambiguity. In order to be consistent with the work order page, I suggest storing this response in the layout.